src/tf-types.ts

```diff
@@ -0,0 +1,173 @@
+import { SupportTable } from './factories/factory-types'
```

Comment:
RDF/JS already has types: https://github.com/rdfjs-base/data-model/blob/master/index.d.ts
Shall we just use those?

Reply:
I wasn't aware of that implementation. I don't know what the consequences are of replacing our internal types with the ones from this one. I think, since that library is just `js` with one (minimal) typings file, that we might lose quite a bit of type information.

Reply:
Not the implementation; just the types.

Reply:
The types are just nine lines of code, they lack pretty much all the methods (which reside in the .js files), or am I missing something?

Reply:
Looks like it's rather the imported `rdf-js` (https://www.npmjs.com/package/@types/rdf-js) is what we want.

Reply:
Ah, these map pretty much 1 on 1. I'll try to replace them.

Reply:
I +1 the use of https://www.npmjs.com/package/@types/rdf-js

Reply:
Turns out it's a bit of a problem to use these types. See #374